relay-server/src/services/store.rs

```diff
@@ -891,6 +908,66 @@ impl StoreService {
         Ok(())
     }
 
+    fn produce_replay_video(
+        &self,
+        event_id: Option<EventId>,
+        scoping: Scoping,
+        payload: Bytes,
+        start_time: Instant,
+        retention: u16,
+    ) -> Result<(), StoreError> {
+        #[derive(Deserialize)]
+        struct VideoEvent<'a> {
+            replay_event: &'a [u8],
+            replay_recording: &'a [u8],
+            replay_video: &'a [u8],
+        }
+
+        let Ok(VideoEvent {
+            replay_video,
+            replay_event,
+            replay_recording,
+        }) = rmp_serde::from_slice::<VideoEvent>(&payload)
+        else {
+            self.outcome_aggregator.send(TrackOutcome {
+                category: DataCategory::Replay,
+                event_id,
+                outcome: Outcome::Invalid(DiscardReason::InvalidReplayEvent),
+                quantity: 1,
+                remote_addr: None,
+                scoping,
+                timestamp: instant_to_date_time(start_time),
+            });
+            return Ok(());
+        };
+
+        self.produce_replay_event(
+            event_id.ok_or(StoreError::NoEventId)?,
+            scoping.organization_id,
+            scoping.project_id,
+            start_time,
+            retention,
+            Bytes::copy_from_slice(replay_event),
+        )?;
+
+        self.produce_replay_recording(
+            event_id,
+            scoping,
+            replay_recording,
+            Some(replay_event),
+            Some(replay_video),
+            start_time,
+            retention,
+        )?;
+
+        metric!(
+            counter(RelayCounters::ProcessingMessageProduced) += 1,
+            event_type = "replay_video"
+        );
```

Comment:
`produce_replay_recording` already logs a metric. We should emit 1 metric for every Kafka message, this would emit two IIURC. What we could do is change the "event type" in `produce_replay_recording` to `recording_with_video` or similar if a video is passed with it.